test_script.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+MODULE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+# Used to specify alternate cache path, useful if run in a docker container
+# NOTE raw datasets will break if you try to transfer the cache from your host to an image
+LM_HARNESSS_CACHE_PATH = os.getenv("LM_HARNESSS_CACHE_PATH")
+
+import torch
+from transformers import (
+    pipeline as trans_pipeline,
+)
+import torch
+
+from lm_eval import simple_evaluate, tasks as tasks_class
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+model = "gpt2"
+
+local_models_path = os.getenv("MODELS_PATH")
+
+task = "text-generation"
+
+print("\n\n\n")
+
+print("Model is: ", model)
+
+print("This task is: ", task)
+
+
+def run_model(model: str, device: str):
+    trans_pipe = trans_pipeline(
+        task=task, model=model, device=device, trust_remote_code=True
+    )
+
+    model = trans_pipe.model
+    tokenizer = trans_pipe.tokenizer
+
+    tasks = [
+        # "squadv2",
+        # "swag",
+        "hellaswag",
+        "lambada_openai",
+        # "glue",
+        # "super-glue-lm-eval-v1",
+        # "wikitext",
+        # "winogrande"
+    ]
+
+    eval_data = simple_evaluate(
+        model="hf-auto",
+        model_args={
+            "pretrained": model,
+            "tokenizer": tokenizer,
+        },
+        limit=0.01,
+        device=device,
+        cache_requests=True,
+        rewrite_requests_cache=True,
+        delete_requests_cache=True,
+        tasks=tasks,
+        write_out=True,
+    )
+
+    return eval_data
+
+
+if __name__ == "__main__":
```

Comment:
Thanks for adding this test script! Would you be willing to add a variant of this to the `tests/` dir for use in our pytest unit tests?

Ideally we'd also have a unit test for `delete_cache()`, `save_to_cache()`, and `load_from_cache()` too if possible, plus potentially a unit test for `Task.build_all_requests()`.

Reply:
Absolutely! I'll got ahead and add it, and then build a unit test for the caching functionality.

Not entirely sure how I'd write one for Task.build_all_requests(), what do you have in mind for it?

Reply:
Alright, it's been moved over into its own script.

I believe I didn't make any changes to Task.build_all_requests(), please correct me if I'm wrong, but may be interested in adding such tests if we end up adding caching to task building as well/might do it for fun later anyway.

The unit tests are now added, and I've confirmed all tests pass on my machine